backend/zoho/zoho.go

```diff
@@ -0,0 +1,1223 @@
+		_, err = f.rename(ctx, srcID, dstLeaf)
+		return err
+	}
+
+	// do the move
+	_, err = f.move(ctx, srcID, dstDirectoryID)
+	if err != nil {
+		return errors.Wrap(err, "couldn't dir move")
+	}
+
+	// Can't copy and change name in one step so we have to check if we have
+	// the correct name after copy
+	if srcLeaf != dstLeaf {
+		_, err = f.rename(ctx, srcID, dstLeaf)
+		if err != nil {
+			return errors.Wrap(err, "dirmove: couldn't rename moved dir")
+		}
+	}
+	srcFs.dirCache.FlushDir(srcRemote)
+	return nil
+}
+
+// About gets quota information
+func (f *Fs) About(ctx context.Context) (usage *fs.Usage, err error) {
+	info, err := f.readMetaDataForID(ctx, f.opt.RootFolderID)
+	if err != nil {
+		return nil, err
+	}
+	usage = &fs.Usage{
+		Used: fs.NewUsageValue(info.Attributes.StorageInfo.Size),
+	}
+	return usage, nil
+}
+
+// DirCacheFlush resets the directory cache - used in testing as an
+// optional interface
+func (f *Fs) DirCacheFlush() {
+	f.dirCache.ResetRoot()
+}
+
+// ------------------------------------------------------------
+
+// Fs returns the parent Fs
+func (o *Object) Fs() fs.Info {
+	return o.fs
+}
+
+// Return a string version
+func (o *Object) String() string {
+	if o == nil {
+		return "<nil>"
+	}
+	return o.remote
+}
+
+// Remote returns the remote path
+func (o *Object) Remote() string {
+	return o.remote
+}
+
+// Storable returns a boolean showing whether this object storable
+func (o *Object) Storable() bool {
+	return true
+}
+
+// Hash returns the SHA-1 of an object returning a lowercase hex string
+func (o *Object) Hash(ctx context.Context, t hash.Type) (string, error) {
+	return "", nil
+}
+
+// Size returns the size of an object in bytes
+func (o *Object) Size() int64 {
+	if err := o.readMetaData(context.TODO()); err != nil {
+		fs.Logf(o, "Failed to read metadata: %v", err)
+		return 0
+	}
+	return o.size
+}
+
+// setMetaData sets the metadata from info
+func (o *Object) setMetaData(info *api.Item) (err error) {
+	if info.Attributes.IsFolder {
+		return fs.ErrorNotAFile
+	}
+	o.hasMetaData = true
+	o.size = info.Attributes.StorageInfo.Size
+	o.modTime = time.Time(info.Attributes.ModifiedTime)
+	o.id = info.ID
+	return nil
+}
+
+// readMetaData gets the metadata if it hasn't already been fetched
+//
+// it also sets the info
+func (o *Object) readMetaData(ctx context.Context) (err error) {
+	if o.hasMetaData {
+		return nil
+	}
+	info, err := o.fs.readMetaDataForPath(ctx, o.remote)
+	if err != nil {
+		return err
+	}
+	return o.setMetaData(info)
+}
+
+// ModTime returns the modification time of the object
+//
+//
+// It attempts to read the objects mtime and if that isn't present the
+// LastModified returned in the http headers
+func (o *Object) ModTime(ctx context.Context) time.Time {
+	return o.modTime
+}
+
+// SetModTime sets the modification time of the local fs object
+func (o *Object) SetModTime(ctx context.Context, modTime time.Time) error {
+	return fs.ErrorCantSetModTime
+}
```

Comment:
Zoho has no way at all to write modification times, neither during upload nor as a separate API call.

Reply:
:-( That is a shame... Is there some metadata we could use?

Reply:
@ncw
Are there other backends without modtime? Are they marked by a feature flag?
We will need modtime for a future chunker garbage collector.

Reply:
@ivandeex there are a few. They are listed in `docs/content/overview.md`. I don't think there is a feature flag right now.

Reply:
You can tell backends without modtime by looking at the `Precision()` returned from the Fs. Though a feature flag isn't a bad idea too.

Custom metadata is a two edged sword - it can be great, but then it is something potentially non-standard. Does the metadata get returned in the listings? If it does then there won't be a performance impact reading it. If you have to query each object to read the metadata then it might not be worth doing.